### PR TITLE
Move Each Table to Own Class with constants and dynamic SQL Generation

### PR DESCRIPTION
**Pull request by cjm721**

### CODE REVIEW

File: src/main/java/ledger/database/storage/ISQLiteDatabaseNote.java [hunk 1]
```diff
@@ -58,14 +62,14 @@ default void editNote(Note note) {
     default List<Note> getAllNotes() throws StorageException {
         try {
             Statement stmt = getDatabase().createStatement();
-            ResultSet rs = stmt.executeQuery("SELECT * FROM NOTE;");
+            ResultSet rs = stmt.executeQuery("SELECT * FROM " + NoteTable.TABLE_NAME + ";");
```
Comment (LadyMopar) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseNote.java]: Should we try to get rid of SELECT * here as well?

File: src/main/java/ledger/database/storage/ISQLiteDatabaseAccount.java [hunk 2]
```diff
@@ -62,15 +66,15 @@ default void editAccount(Account account) throws StorageException {
     default List<Account> getAllAccounts() throws StorageException {
         try {
             Statement stmt = getDatabase().createStatement();
-            ResultSet rs = stmt.executeQuery("SELECT * FROM ACCOUNT;");
+            ResultSet rs = stmt.executeQuery("SELECT * FROM " + AccountTable.TABLE_NAME + ";");
```
Comment (LadyMopar) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseAccount.java]: SELECT * ?

File: src/main/java/ledger/database/storage/table/TypeTable.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            "TYPE_NAME TEXT              NOT NULL, " +
+            "TYPE_DESC TEXT              NOT NULL" +
+            ")";
+
+    public static String TABLE_NAME = "TYPE";
+    public static String TYPE_ID = "TYPE_ID";
+    public static String TYPE_NAME = "TYPE_NAME";
+    public static String TYPE_DESC = "TYPE_DESC";
+
+    public static String CreateStatement() {
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/table/TypeTable.java]: public methods need a Javadoc

File: src/main/java/ledger/database/storage/table/TransactionTable.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    public static final String TABLE_NAME = "TRANSACT";
+    public static final String TRANS_ID = "TRANS_ID";
+    public static final String TRANS_DATETIME = "TRANS_DATETIME";
+    public static final String TRANS_AMOUNT = "TRANS_AMOUNT";
+    public static final String TRANS_PENDING = "TRANS_PENDING";
+    public static final String TRANS_ACCOUNT_ID = "TRANS_ACCOUNT_ID";
+    public static final String TRANS_PAYEE_ID = "TRANS_PAYEE_ID";
+    public static final String TRANS_TYPE_ID = "TRANS_TYPE_ID";
+
+    public static String CreateStatement() {
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/table/TransactionTable.java]: Needs Javadoc

File: src/main/java/ledger/database/storage/table/TagToTransTable.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+            "TTTS_TRANS_ID INT          NOT NULL, " +
+            "FOREIGN KEY(TTTS_TAG_ID) REFERENCES TAG(TAG_ID), " +
+            "FOREIGN KEY(TTTS_TRANS_ID) REFERENCES TRANSACT(TRANS_ID)" +
+            ")";
+
+    public static String TABLE_NAME = "TAG_TO_TRANS";
+    public static String TTTS_TAG_ID = "TTTS_TAG_ID";
+    public static String TTTS_TRANS_ID = "TTTS_TRANS_ID";
+
+    public static String CreateStatement() {
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/table/TagToTransTable.java]: Javadoc

File: src/main/java/ledger/database/storage/table/TagToPayeeTable.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+            "TTPE_PAYEE_ID INT          NOT NULL, " +
+            "FOREIGN KEY(TTPE_TAG_ID) REFERENCES TAG(TAG_ID), " +
+            "FOREIGN KEY(TTPE_PAYEE_ID) REFERENCES PAYEE(PAYEE_ID)" +
+            ")";
+
+    public static String TABLE_NAME = "TAG_TO_PAYEE";
+    public static String TTPE_TAG_ID = "TTPE_TAG_ID";
+    public static String TTPE_PAYEE_ID = "TTPE_PAYEE_ID";
+
+    public static String CreateStatement() {
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/table/TagToPayeeTable.java]: Javadoc

File: src/main/java/ledger/database/storage/table/TagTable.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+            "TAG_NAME TEXT              NOT NULL, " +
+            "TAG_DESC TEXT              NOT NULL" +
+            ")";
+
+    public static String TABLE_NAME = "TAG";
+    public static String TAG_ID = "TAG_ID";
+    public static String TAG_NAME = "TAG_NAME";
+    public static String TAG_DESC = "TAG_DESC";
+
+    public static String CreateStatement() {
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/table/TagTable.java]: Javadoc

File: src/main/java/ledger/database/storage/table/PayeeTable.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+            "PAYEE_NAME TEXT           NOT NULL, " +
+            "PAYEE_DESC TEXT           NOT NULL" +
+            ")";
+
+    public static String TABLE_NAME = "PAYEE";
+    public static String PAYEE_ID = "PAYEE_ID";
+    public static String PAYEE_NAME = "PAYEE_NAME";
+    public static String PAYEE_DESC = "PAYEE_DESC";
+
+    public static String CreateStatement() {
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/table/PayeeTable.java]: Needs a Javadoc

File: src/main/java/ledger/database/storage/table/NoteTable.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+            "(NOTE_TRANS_ID INTEGER PRIMARY KEY AUTOINCREMENT, " +
+            "NOTE_TEXT TEXT             NOT NULL, " +
+            "FOREIGN KEY(NOTE_TRANS_ID) REFERENCES TRANSACT(TRANS_ID)" +
+            ")";
+
+    public static final String TABLE_NAME = "NOTE";
+    public static final String NOTE_TRANS_ID = "NOTE_TRANS_ID";
+    public static final String NOTE_TEXT = "NOTE_TEXT";
+
+    public static String CreateStatement() {
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/table/NoteTable.java]: Javadoc

File: src/main/java/ledger/database/storage/table/AccountTable.java [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+            "ACCOUNT_DESC TEXT           NOT NULL" +
+            ")";
+
+    public static final String TABLE_NAME = "ACCOUNT";
+
+    public static final String ACCOUNT_ID = "ACCOUNT_ID";
+    public static final String ACCOUNT_NAME = "ACCOUNT_NAME";
+    public static final String ACCOUNT_DESC = "ACCOUNT_DESC";
+
+    public static String CreateStatement() {
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/table/AccountTable.java]: Javadoc

File: src/main/java/ledger/database/storage/table/AccountBalanceTable.java [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+            "ABAL_DATETIME REAL         NOT NULL, " +
+            "ABAL_AMOUNT INT            NOT NULL " +
+            ")";
+
+    public static String TABLE_NAME = "ACCOUNT_BALANCE";
+    public static String ABAL_ACCOUNT_ID = "ABAL_ACCOUNT_ID";
+    public static String ABAL_DATETIME = "ABAL_DATETIME";
+    public static String ABAL_AMOUNT = "ABAL_AMOUNT";
+
+    public static String CreateStatement() {
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/table/AccountBalanceTable.java]: Needs a javadoc

File: src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java [hunk 12]
```diff
@@ -408,7 +435,8 @@ default Account getAccountForID(int accountID) throws StorageException {
 
     default Type getTypeForID(int typeID) throws StorageException {
         try {
-            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM TYPE WHERE TYPE_ID=?");
+            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM " + TypeTable.TABLE_NAME +
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java]: Lets get rid of SELECT * like Sriram was advising

File: src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java [hunk 13]
```diff
@@ -395,7 +421,8 @@ default Payee getPayeeForID(int payeeID) throws StorageException {
 
     default Account getAccountForID(int accountID) throws StorageException {
         try {
-            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM ACCOUNT WHERE ACCOUNT_ID=?");
+            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM " + AccountTable.TABLE_NAME +
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java]: Lets get rid of SELECT *, per sriram's advice

File: src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java [hunk 14]
```diff
@@ -376,14 +401,15 @@ default Note getNoteForTransactionID(int transactionID) throws StorageException
 
     default Payee getPayeeForID(int payeeID) throws StorageException {
         try {
-            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM PAYEE WHERE PAYEE_ID=?");
+            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM " + PayeeTable.TABLE_NAME +
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java]: Lets get rid of SELECT * like Sriram was advised

File: src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java [hunk 15]
```diff
@@ -345,22 +368,24 @@ default Note getNoteForTransactionID(int transactionID) throws StorageException
...
             stmt.setInt(1, transactionID);
 
             ResultSet rs = stmt.executeQuery();
 
             while (rs.next()) {
-                int tagId = rs.getInt("TTTS_TAG_ID");
+                int tagId = rs.getInt(TagToTransTable.TTTS_TAG_ID);
 
-                PreparedStatement tagStmt = getDatabase().prepareStatement("SELECT * FROM TAG WHERE TAG_ID = ?");
+                PreparedStatement tagStmt = getDatabase().prepareStatement("SELECT * FROM " + TagTable.TABLE_NAME +
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java]: Lets get rid of SELECT * like Sriram was advised

File: src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java [hunk 16]
```diff
@@ -345,22 +368,24 @@ default Note getNoteForTransactionID(int transactionID) throws StorageException
         List<Tag> tags = new ArrayList<>();
 
         try {
-            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM TAG_TO_TRANS WHERE TTTS_TRANS_ID=?");
+            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM " + TagToTransTable.TABLE_NAME +
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java]: Lets get rid of SELECT * like Sriram was advised

File: src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java [hunk 17]
```diff
@@ -324,13 +346,14 @@ default void insertTagToTrans(int tagID, int transID) {
 
     default Note getNoteForTransactionID(int transactionID) throws StorageException {
         try {
-            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM NOTE WHERE NOTE_TRANS_ID=?");
+            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM " + NoteTable.TABLE_NAME +
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java]: Lets get rid of SELECT * like Sriram was advised

File: src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java [hunk 18]
```diff
@@ -290,16 +310,17 @@ default Tag getTagForNameAndDescription(String tagName, String tagDescription) {
 
     default Payee getPayeeForNameAndDescription(String payeeName, String payeeDescription) {
         try {
-            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM PAYEE WHERE PAYEE_NAME=? AND PAYEE_DESC=?");
+            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM " + PayeeTable.TABLE_NAME +
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java]: Lets get rid of SELECT * like Sriram advised

File: src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java [hunk 19]
```diff
@@ -256,7 +274,8 @@ default void deleteAllTagToTransForTransactionID(int transactionID) throws Stora
 
     default Account getAccountForName(String name) throws StorageException {
         try {
-            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM ACCOUNT WHERE ACCOUNT_NAME=?");
+            PreparedStatement stmt = getDatabase().prepareStatement("SELECT * FROM " + AccountTable.TABLE_NAME +
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java]: Lets get rid of SELECT * like Sriram advised

File: src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java [hunk 20]
```diff
@@ -223,7 +238,8 @@ default Type getTypeForName(String name) throws StorageException {
         ResultSet rs;
 
         try {
-            stmt = getDatabase().prepareStatement("SELECT * FROM TYPE WHERE TYPE_NAME=?");
+            stmt = getDatabase().prepareStatement("SELECT * FROM " + TypeTable.TABLE_NAME +
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java]: Lets get rid of SELECT * like Sriram advised

File: src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java [hunk 21]
```diff
@@ -187,18 +202,18 @@ default void editTransaction(Transaction transaction) throws StorageException {
     default List<Transaction> getAllTransactions() throws StorageException {
         try {
             Statement stmt = getDatabase().createStatement();
-            ResultSet rs = stmt.executeQuery("SELECT * FROM TRANSACT;");
+            ResultSet rs = stmt.executeQuery("SELECT * FROM " + TransactionTable.TABLE_NAME + ";");
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java]: Lets get rid of SELECT * like Sriram advised

File: src/main/java/ledger/database/storage/ISQLiteDatabaseTag.java [hunk 22]
```diff
@@ -61,15 +64,15 @@ default void editTag(Tag tag) throws StorageException {
     default List<Tag> getAllTags() throws StorageException {
         try {
             Statement stmt = getDatabase().createStatement();
-            ResultSet rs = stmt.executeQuery("SELECT * FROM TAG;");
+            ResultSet rs = stmt.executeQuery("SELECT * FROM " + TagTable.TABLE_NAME + ";");
```
Comment (shellajt) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseTag.java]: Lets get rid of SELECT * like Sriram advised

File: src/main/java/ledger/database/storage/ISQLiteDatabaseNote.java (hunk 1 above)
Comment (taylerhow) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseNote.java]: I got rid of all occurrences of "SELECT *" in the ISQLiteDatabase____ files in my refactoring work on the f_2refactoring branch.

File: src/main/java/ledger/database/storage/ISQLiteDatabaseAccount.java (hunk 2 above)
Comment (taylerhow) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseAccount.java]: I got rid of all occurrences of "SELECT *" in the ISQLiteDatabase____ files in my refactoring work on the f_2refactoring branch.

File: src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java (hunk 21 above)
Comment (taylerhow) [inline src/main/java/ledger/database/storage/ISQLiteDatabaseTransaction.java]: I got rid of all occurrences of "SELECT *" in the ISQLiteDatabase____ files in my refactoring work on the f_2refactoring branch. These changes might not merge nicely, though. (Applies to all other "SELECT *" change requests, I believe)

**Pull request merged**